Src/ILGPU/Context.cs

```diff
@@ -384,6 +385,77 @@ public DeviceCollection<TDevice> GetDevices<TDevice>()
                 : new DeviceCollection<TDevice>(new List<Device>());
         }
 
+        /// <summary>
+        /// Attempts to return the most optimal single device.
+        /// </summary>
+        /// <param name="PreferCPU">Always returns CPU device 0.</param>
+        /// <returns>Selected device.</returns>
+        public Device GetBestDevice(bool PreferCPU = false)
```

Comment:
nit: we do not use default parameters, since they can break ABI compatibility. We have to create additional overloads that specify the parameter values. Moreover, all parameters name should be written in `CamelCase`.